org.eclipse.jdt.core/codeassist/org/eclipse/jdt/internal/codeassist/CompletionEngine.java

```diff
@@ -6091,6 +6117,7 @@ private void findConstructors(
 								continue next;
 						}
 
+					constructorsFound.add(new Object[]{constructor, currentType});
```

Comment:
Passing this looks weird, but considering the is 400+ lines long, I'm not sure the code gets much worse.

Though is it really necessary to gather the list and then add suggestions for it? You cannot add suggestions right here?

Reply:
This collected constructors are not used for creating completions. They are used for other purposes such as eliminating duplicate constructors and also trying to figure our parameter types. It is same as methodsFound and typesFound collections in other methods.